src/util/ieee_float.cpp

```diff
@@ -1193,10 +1199,9 @@ double ieee_floatt::to_double() const
 /// for NaN.
 float ieee_floatt::to_float() const
 {
-  if(sizeof(unsigned)!=sizeof(float))
-  {
-    throw "ieee_floatt::to_float not supported on this architecture";
-  }
+  // if false - ieee_floatt::to_float not supported on this architecture
+  static_assert(sizeof(unsigned)==sizeof(float),
```

Comment:
Taking another look at this this should actually be `sizeof(float) == sizeof(uint32_t)` and the message should say that _float_ should be 32 bits, not the integer